lib/worker/worker.js

```diff
@@ -0,0 +1,210 @@
+    ),
+  };
+}
+
+async function fixFile (meta, filePath, fileContent) {
+  const { ESLint, eslintFix } = meta;
+  const results = await lintPathOrContent(eslintFix, filePath, fileContent);
+  await ESLint.outputFixes(results);
+  return results;
+}
+
+function countMessages (results) {
+  let count = 0;
+  for (let { messages } of results) {
+    count += messages.length;
+  }
+  return count;
+}
+
+async function jobESLintFix ({
+  projectPath, config, filePath, isModified, contents
+}) {
+  const eslint = getESLint(filePath, config, { projectPath });
+  process.chdir(eslint.cwd);
+
+  const lintResults = await lintFile(eslint, filePath, contents);
+  const lintMessageCount = countMessages(lintResults);
+
+  const fixResults = await fixFile(eslint, filePath, contents);
+  const rulesMeta = eslint.eslintFix.getRulesMetaForResults(fixResults);
+
+  const result = {
+    rules: rulesMeta,
+    results: convertResults(
+      fixResults,
+      rulesMeta,
+      config,
+      { isModified }
+    ),
+  };
+
+  if (lintMessageCount > 0) {
+    result.fixCount = lintMessageCount - result.results.length;
+  }
+
+  return result;
+}
+
+async function processMessage (bundle) {
+  const oldCwd = process.cwd();
+
+  try {
+    const { type } = bundle;
+    if (type == null) {
+      throw new JobTypeError();
+    }
+
+    switch (type) {
+      case 'clear-cache': return jobClearCache(bundle);
+      case 'debug': return jobDebug(bundle);
+      case 'lint': return jobESLintLint(bundle);
+      case 'fix': return jobESLintFix(bundle);
+      default: throw new JobNotFoundError(type);
+    }
+  } finally {
+    process.chdir(oldCwd);
+  }
+}
+
+function handleError(error, { key, filePath, projectPath } = {}) {
+  if (error.messageTemplate === 'no-config-found') {
+    error = new ConfigNotFoundError(error.message, filePath, projectPath);
+  }
+
+  if (error instanceof Error) {
+    error.error = error.message;
+    error.key = key;
+
+    return error;
+  }
+
+  return new UnknownError();
+}
+
+if (require.main === module) {
+  process.stdin
+  .pipe(ndjson.parse({ strict: false }))
+  .on('data', (data) => {
+    Messaging.runWith(data, async () => {
+      try {
+        if (!data.key) {
+          throw new JobKeyError();
+        }
+
+        Messaging.emit(await processMessage(data));
+      } catch (error) {
+        Messaging.emitError(handleError(error, data));
+      }
+    });
+  });
+
+  process.stdin.resume();
+
+  process.on('uncaughtException', (error) => {
+    error.uncaught = true;
+    error.error = 'Unknown error';
+
+    // We're catching this exception so that we can try to emit it as JSON
+    // before exiting. That way `JobManager` can at least attempt to report
+    // something meaningful to the user. But emitting the error as JSON might
+    // _itself_ fail, so we've got to guard against that, lest we start an
+    // infinite loop.
+    try {
+      Messaging.emitError(error);
+    } finally {
+      process.exit(1);
+    }
```

Comment:
I am not sure we need this anymore, but I wanted to leave this in as I don't want to find out the hard way with a ticket later 😂